File: src/main/java/com/onmeet/company/repository/CompanyRepository.java
package com.onmeet.company.repository;

import com.onmeet.company.entity.Company;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CompanyRepository extends JpaRepository<Company, String> {
    Optional<Company> findByDomain(String domain);
}

File: auth-service/src/main/kotlin/com/onmeet/auth/repository/jpa/CompanyRepository.kt
package com.onmeet.auth.repository.jpa

import com.onmeet.auth.entity.Company
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CompanyRepository : JpaRepository<Company, Long> {
    fun findByName(name: String): Company?
}
